System.Device.I2c/I2cDevice.cs

```diff
@@ -0,0 +1,135 @@
+        /// </summary>
+        /// <param name="buffer">
+        /// The buffer that contains the data to be written to the I2C device.
+        /// The data should not include the I2C device address.
+        /// </param>
+        public void Write(SpanByte buffer)
+        {
+            // This is allocating an intermediate buffer using the buffer of 
+            // the SpanByte. This is intend to be changed in a native implementation
+            _device.Write(buffer.ToArray());
+        }
+
+        /// <summary>
+        /// Performs an atomic operation to write data to and then read data from the I2C bus on which the device is connected,
+        /// and sends a restart condition between the write and read operations.
+        /// </summary>
+        /// <param name="writeBuffer">
+        /// The buffer that contains the data to be written to the I2C device.
+        /// The data should not include the I2C device address.</param>
+        /// <param name="readBuffer">
+        /// The buffer to read the data from the I2C device.
+        /// The length of the buffer determines how much data to read from the I2C device.
+        /// </param>
+        public void WriteRead(SpanByte writeBuffer, SpanByte readBuffer)
+        {
+            // This is allocating an intermediate buffer and then copy back the data to 
+            // the SpanByte. This is intend to be changed in a native implementation
+            byte[] toRead = new byte[readBuffer.Length];
+            _device.WriteRead(writeBuffer.ToArray(), toRead);
+            for (int i = 0; i < toRead.Length; i++)
+            {
+                readBuffer[i] = toRead[i];
+            }
+        }
+
+        /// <summary>
+        /// Creates a communications channel to a device on an I2C bus running on the current platform
+        /// </summary>
+        /// <param name="settings">The connection settings of a device on an I2C bus.</param>
+        /// <returns>A communications channel to a device on an I2C bus running on Windows 10 IoT.</returns>
+        public static I2cDevice Create(I2cConnectionSettings settings)
+        {
+            return new I2cDevice(settings);
+        }
+
+        /// <summary>
+        /// Create an I2C Device
+        /// </summary>
+        /// <param name="settings">Connection settings</param>
+        public I2cDevice(I2cConnectionSettings settings)
+        {
+            ConnectionSettings = settings;
+            _device = Windows.Devices.I2c.I2cDevice.FromId($"I2C{settings.BusId}", new Windows.Devices.I2c.I2cConnectionSettings(settings.DeviceAddress)
+            {
+                BusSpeed = (Windows.Devices.I2c.I2cBusSpeed)settings.BusSpeed,
+                SharingMode = (Windows.Devices.I2c.I2cSharingMode)settings.SharingMode
+            });
+        }
+
+        /// <inheritdoc cref="IDisposable.Dispose"/>
+        public void Dispose()
+        {
+            Dispose(true);
+            GC.SuppressFinalize(this);
```

Comment:
why do you suppress finalizer if this is no-op? Should other Dispose be virtual?

Reply:
I just reused the way it was. Happy to change and adjust anything. Maybe @josesimoes has some ideas about it and there may be reasons why.
Note that the W.D.I2c I2c controller has no dispose

Reply:
This was added on most System.Devices classes that required native dispose to free resources and that's the reason it's showing here. `GC.SuppressFinalize` it's fully implemented in nF and doing what is supposed to do.